packages/components/src/calendar/date-range-calendar/index.stories.tsx

```diff
@@ -309,90 +253,325 @@ const lastTwelveMonths = ( date: Date ) => {
+						flex-shrink: 0;
+						padding: 12px 16px;
+						border-inline-end: 1px solid #ccc;
+					}
+					.preset-listbox:has([data-active-item]):focus {
+						outline: 3px solid transparent;
+					}
+					.preset-listbox__group {
+						outline: 1px solid red;
+					}
+					.preset-listbox__item:last-child {
+						margin-block-start: auto;
+					}
+					.preset-listbox:focus .preset-listbox__item[data-active-item] {
+						box-shadow:0 0 0 var(--wp-admin-border-width-focus) var(--wp-components-color-accent, var(--wp-admin-theme-color, #3858e9));
+						outline: 3px solid transparent;
+					}
+					.preset-listbox__item-label {
+						width: 100%;
+						display: flex;
+						align-items: center;
+						gap: 8px;
+						justify-content: space-between;
+					}
+					.preset-listbox__item-label-bullet {
+						font-size: 2em;
+						position: relative;
+						top: -0.06em;
+					}
+					.date-controls-wrapper {
+						padding: 24px;
+					}
+					.date-input {
+						flex: 1;
+					}
+					.date-range-calendar {
+						z-index: 0;
+						min-height: 300px;
+					}
+					.reset-button {
+						position: absolute;
+						inset-inline-end: 24px;
+						inset-block-end: 12px;
+					}
+				` }</style>
+				<Button
+					__next40pxDefaultSize
+					variant="secondary"
+					ref={ popoverTriggerRef }
+					onClick={ () => setIsPopoverOpen( ! isPopoverOpen ) }
+				>
+					{ ! range || ! range.from
+						? 'Pick a date range'
+						: `Selected:${ range.from.toLocaleDateString() } - ${
+								range.to?.toLocaleDateString() ?? '?'
+						  }` }
+				</Button>
+
+				{ isPopoverOpen && (
+					<Popover
+						role="dialog"
+						open={ isPopoverOpen }
+						offset={ 16 }
+						anchor={ popoverTriggerRef.current }
+						aria-labelledby={ popoverTitleId }
 					>
-						Year to date
-					</button>
-				</div>
-				<DateRangeCalendar
-					{ ...args }
-					selected={ range }
-					onSelect={ ( selectedDate, ...rest ) => {
-						setRange(
-							// Set controlled state to null if there's no selection
-							! selectedDate || ( selectedDate.from === undefined && selectedDate.to === undefined )
-								? null
-								: selectedDate
-						);
-						args.onSelect?.( selectedDate, ...rest );
-					} }
-					month={ month }
-					onMonthChange={ setMonth }
-				/>
+						<VisuallyHidden id={ popoverTitleId }>Select a date range</VisuallyHidden>
+						<HStack spacing={ 0 } className="popover-content-wrapper" alignment="top">
+							<VisuallyHidden id={ presetListboxId }>Date range presets</VisuallyHidden>
+							<Composite
+								aria-labelledby={ presetListboxId }
+								activeId={ activePresetId }
+								setActiveId={ ( activeId ) => {
+									setActivePresetId( activeId ?? null );
+								} }
+								focusLoop
+								virtualFocus
+								render={
+									// @ts-expect-error children are passed directly to Composite
+									<VStack justify="flex-start" alignment="stretch" spacing={ 0 } />
+								}
+								role="listbox"
```

Comment:
Working on this example was a great opportunity to investigate how we could potentially implement an inline [`listbox` pattern](https://www.w3.org/WAI/ARIA/apg/patterns/listbox/) (including semantics and keyboard behaviour) (cc @jameskoster @mirka )